### DISCUSSION TITLE
Cannot read property 'map' of undefined when using regexManagers

### DISCUSSION THREAD

**Discussion by rubasace** [renovatebot/renovate#16124] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate/renovate:32.76.0@sha256:4852438101c0d7c2e35144c724c6865d179b0d014f8b83a7afec06066952c52e

### Please select which platform you are using if self-hosting.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the bug

I recently started to get the `Cannot read property 'map' of undefined` error on a repository that uses `regexManagers`. 

We have these lines in our Dockerfile:

```Dockerfile
# Workaround to keep sdk version up to date: docker image version is aligned with the SDK one, so we can trick Renovate telling it that the SDK_VERSION ARG is the docker image
# renovate: depName=google/cloud-sdk
ARG SDK_VERSION=388.0.0
``` 

And we have configured the `renovate.json` file to use a custom manager that reads that comment:

```js
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    "docker:enableMajor",
    "docker:pinDigests",
    "default:automergeDigest",
    "default:automergePatch"
  ],
  "automergeStrategy": "squash",
  "prHourlyLimit": "6", 
  "reviewers": ["****", "****", "****", "****"],
  "packageRules": [
    {
      "matchUpdateTypes": ["pin"],
      "automerge": true
    }
  ],
  "regexManagers": [
    {
      "fileMatch": ["^Dockerfile$"],
      "matchStrings": [
        "renovate: depName=(?<depName>.*?)\\s*?ARG\\s*?.*?_VERSION=(?<currentValue>.*)"
      ],
      "datasourceTemplate": "docker",
      "versioningTemplate": "semver"
    }
  ],
  "packageRules": [
    {
      "matchPackageNames": ["renovate/renovate", "google/cloud-sdk"],
      "automerge": true,
      "separateMinorPatch": false
    }
  ]
}
```

We do it this way cause the google/cloud-sdk Docker image uses the same versioning as the CLI tool, so we can keep our custom image up to date, generating a new one every time the tool is updated. 

This approached worked till a couple weeks ago: having a look at our commits, I can see the last PR was done on the 1st of June, updating to version `388.0.0`. Since then, two new versions have been released but no PRs, and that issue is all I can see on my logs.

### Relevant debug logs

<details><summary>Logs</summary>

```json
{
  "name": "renovate",
  "hostname": "****",
  "pid": 35,
  "level": 50,
  "logContext": "wJvOt_Ui8MAUhPWp2PpE9",
  "repository": "****",
  "err": {
    "message": "Cannot read property 'map' of undefined",
    "stack": "TypeError: Cannot read property 'map' of undefined\n    at handleAny (/usr/src/app/node_modules/renovate/lib/modules/manager/regex/strategies.ts:17:6)\n    at Object.extractPackageFile (/usr/src/app/node_modules/renovate/lib/modules/manager/regex/index.ts:27:23)\n    at extractPackageFile (/usr/src/app/node_modules/renovate/lib/modules/manager/index.ts:70:9)\n    at getManagerPackageFiles (/usr/src/app/node_modules/renovate/lib/workers/repository/extract/manager-files.ts:66:43)\n    at /usr/src/app/node_modules/renovate/lib/workers/repository/extract/index.ts:50:28\n    at async Promise.all (index 2)\n    at extractAllDependencies (/usr/src/app/node_modules/renovate/lib/workers/repository/extract/index.ts:48:26)\n    at extract (/usr/src/app/node_modules/renovate/lib/workers/repository/process/extract-update.ts:82:20)\n    at extractDependencies (/usr/src/app/node_modules/renovate/lib/workers/repository/process/index.ts:105:26)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:42:52)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:152:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:16:22"
  },
  "msg": "Repository has unknown error",
  "time": "2022-06-17T09:20:06.663Z",
  "v": 0
}
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment, marked as answer]
Please update to the latest version and retry (and in future please do that as a precaution before creating any bug report)

**Reply by rubasace** [reply]
Oh sorry about that... the repository that is having this issue is Renovate's itself, I didn't realize it was not updating itself either and just made the (wrong) assumption that I was on the latest version. 
I'm rebuilding now with latest, I'll let you know in a bit if the issue gets resolved.

**Reply by rarkins** [reply]
Whoops! That unfortunately can happen occasionally when you go into Renovate-updating-Renovate inception mode. Sorry for the inconvenience

**Reply by rubasace** [reply]
Ok, I can confirm it works on latest (`renovate/renovate:32.89.0@sha256:d8d6b975d59895560a3e55322eb61fe81222379a3cb62a8c7913782770d0f0ea`). 

Sorry for the disturbance, I didn't realize we had hit some kind of deadlock :)
